{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df1 = pd.read_csv('../data/genes.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PeakClass</th>\n",
       "      <th>SpecID</th>\n",
       "      <th>PeakTick</th>\n",
       "      <th>PeakMass</th>\n",
       "      <th>SN</th>\n",
       "      <th>MinTick</th>\n",
       "      <th>MaxTick</th>\n",
       "      <th>MinMass</th>\n",
       "      <th>MaxMass</th>\n",
       "      <th>NumSpec</th>\n",
       "      <th>MinSN</th>\n",
       "      <th>MaxSN</th>\n",
       "      <th>MeanSN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>18</td>\n",
       "      <td>3522</td>\n",
       "      <td>964.63</td>\n",
       "      <td>66.20</td>\n",
       "      <td>3522</td>\n",
       "      <td>3526</td>\n",
       "      <td>964.63</td>\n",
       "      <td>967.05</td>\n",
       "      <td>17</td>\n",
       "      <td>40.83</td>\n",
       "      <td>69.93</td>\n",
       "      <td>53.965882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "      <td>3522</td>\n",
       "      <td>964.63</td>\n",
       "      <td>64.70</td>\n",
       "      <td>3522</td>\n",
       "      <td>3526</td>\n",
       "      <td>964.63</td>\n",
       "      <td>967.05</td>\n",
       "      <td>17</td>\n",
       "      <td>40.83</td>\n",
       "      <td>69.93</td>\n",
       "      <td>53.965882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>3523</td>\n",
       "      <td>965.23</td>\n",
       "      <td>69.93</td>\n",
       "      <td>3522</td>\n",
       "      <td>3526</td>\n",
       "      <td>964.63</td>\n",
       "      <td>967.05</td>\n",
       "      <td>17</td>\n",
       "      <td>40.83</td>\n",
       "      <td>69.93</td>\n",
       "      <td>53.965882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>3523</td>\n",
       "      <td>965.23</td>\n",
       "      <td>69.39</td>\n",
       "      <td>3522</td>\n",
       "      <td>3526</td>\n",
       "      <td>964.63</td>\n",
       "      <td>967.05</td>\n",
       "      <td>17</td>\n",
       "      <td>40.83</td>\n",
       "      <td>69.93</td>\n",
       "      <td>53.965882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>3523</td>\n",
       "      <td>965.23</td>\n",
       "      <td>57.67</td>\n",
       "      <td>3522</td>\n",
       "      <td>3526</td>\n",
       "      <td>964.63</td>\n",
       "      <td>967.05</td>\n",
       "      <td>17</td>\n",
       "      <td>40.83</td>\n",
       "      <td>69.93</td>\n",
       "      <td>53.965882</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   PeakClass  SpecID  PeakTick  PeakMass     SN  MinTick  MaxTick  MinMass  \\\n",
       "0          1      18      3522    964.63  66.20     3522     3526   964.63   \n",
       "1          1      17      3522    964.63  64.70     3522     3526   964.63   \n",
       "2          1      10      3523    965.23  69.93     3522     3526   964.63   \n",
       "3          1      20      3523    965.23  69.39     3522     3526   964.63   \n",
       "4          1       5      3523    965.23  57.67     3522     3526   964.63   \n",
       "\n",
       "   MaxMass  NumSpec  MinSN  MaxSN     MeanSN  \n",
       "0   967.05       17  40.83  69.93  53.965882  \n",
       "1   967.05       17  40.83  69.93  53.965882  \n",
       "2   967.05       17  40.83  69.93  53.965882  \n",
       "3   967.05       17  40.83  69.93  53.965882  \n",
       "4   967.05       17  40.83  69.93  53.965882  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2 = pd.read_csv('../data/peakInfo.txt', sep='\\t')\n",
    "df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#OTU ID</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OTU0\\t</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OTU1\\t</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Clostridiales</td>\n",
       "      <td>Ruminococcaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OTU2\\t</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Lactobacillales</td>\n",
       "      <td>Enterococcaceae</td>\n",
       "      <td>Enterococcus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OTU3\\t</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Porphyromonadaceae</td>\n",
       "      <td>Parabacteroides</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OTU4\\t</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #OTU ID\\t   Kingdom         Phylum          Class            Order  \\\n",
       "0    OTU0\\t  Bacteria            NaN            NaN              NaN   \n",
       "1    OTU1\\t  Bacteria     Firmicutes     Clostridia    Clostridiales   \n",
       "2    OTU2\\t  Bacteria     Firmicutes        Bacilli  Lactobacillales   \n",
       "3    OTU3\\t  Bacteria  Bacteroidetes  Bacteroidetes    Bacteroidales   \n",
       "4    OTU4\\t  Bacteria  Bacteroidetes            NaN              NaN   \n",
       "\n",
       "               Family            Genus Unnamed: 7 Unnamed: 8  \n",
       "0                 NaN              NaN        NaN        NaN  \n",
       "1     Ruminococcaceae              NaN        NaN        NaN  \n",
       "2     Enterococcaceae     Enterococcus        NaN        NaN  \n",
       "3  Porphyromonadaceae  Parabacteroides        NaN        NaN  \n",
       "4                 NaN              NaN        NaN        NaN  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3 = pd.read_csv('../data/MicrobiomeOTUtaxonomy.csv')\n",
    "df3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['#OTU ID\\t', 'Kingdom', 'Phylum', 'Class', 'Order', 'Family', 'Genus',\n",
       "       'Unnamed: 7', 'Unnamed: 8'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Strip the trailing TAB from the first column, also in the column header"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#OTU ID</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OTU0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OTU1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Clostridiales</td>\n",
       "      <td>Ruminococcaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OTU2</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Lactobacillales</td>\n",
       "      <td>Enterococcaceae</td>\n",
       "      <td>Enterococcus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OTU3</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Porphyromonadaceae</td>\n",
       "      <td>Parabacteroides</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OTU4</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #OTU ID   Kingdom         Phylum          Class            Order  \\\n",
       "0    OTU0  Bacteria            NaN            NaN              NaN   \n",
       "1    OTU1  Bacteria     Firmicutes     Clostridia    Clostridiales   \n",
       "2    OTU2  Bacteria     Firmicutes        Bacilli  Lactobacillales   \n",
       "3    OTU3  Bacteria  Bacteroidetes  Bacteroidetes    Bacteroidales   \n",
       "4    OTU4  Bacteria  Bacteroidetes            NaN              NaN   \n",
       "\n",
       "               Family            Genus Unnamed: 7 Unnamed: 8  \n",
       "0                 NaN              NaN        NaN        NaN  \n",
       "1     Ruminococcaceae              NaN        NaN        NaN  \n",
       "2     Enterococcaceae     Enterococcus        NaN        NaN  \n",
       "3  Porphyromonadaceae  Parabacteroides        NaN        NaN  \n",
       "4                 NaN              NaN        NaN        NaN  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3.columns = df3.columns.str.strip(\"\\t\")\n",
    "df3[\"#OTU ID\"] = df3[\"#OTU ID\"].str.rstrip(\"\\t\")\n",
    "df3.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove the last two columns. They are empty"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#OTU ID</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OTU0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OTU1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Clostridiales</td>\n",
       "      <td>Ruminococcaceae</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OTU2</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Lactobacillales</td>\n",
       "      <td>Enterococcaceae</td>\n",
       "      <td>Enterococcus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OTU3</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Porphyromonadaceae</td>\n",
       "      <td>Parabacteroides</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OTU4</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #OTU ID   Kingdom         Phylum          Class            Order  \\\n",
       "0    OTU0  Bacteria            NaN            NaN              NaN   \n",
       "1    OTU1  Bacteria     Firmicutes     Clostridia    Clostridiales   \n",
       "2    OTU2  Bacteria     Firmicutes        Bacilli  Lactobacillales   \n",
       "3    OTU3  Bacteria  Bacteroidetes  Bacteroidetes    Bacteroidales   \n",
       "4    OTU4  Bacteria  Bacteroidetes            NaN              NaN   \n",
       "\n",
       "               Family            Genus  \n",
       "0                 NaN              NaN  \n",
       "1     Ruminococcaceae              NaN  \n",
       "2     Enterococcaceae     Enterococcus  \n",
       "3  Porphyromonadaceae  Parabacteroides  \n",
       "4                 NaN              NaN  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "del df3['Unnamed: 7']\n",
    "df3.drop('Unnamed: 8', axis=1, inplace=True)\n",
    "df3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cell type</th>\n",
       "      <th>Evidence type</th>\n",
       "      <th>Feature name</th>\n",
       "      <th>Location</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A549</td>\n",
       "      <td>DNase1 &amp; TFBS</td>\n",
       "      <td>DNase1</td>\n",
       "      <td>13:32315692-32316237</td>\n",
       "      <td>ENCODE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A549</td>\n",
       "      <td>Hists &amp; Pols</td>\n",
       "      <td>H3K4me3</td>\n",
       "      <td>13:32314785-32316918</td>\n",
       "      <td>ENCODE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A549</td>\n",
       "      <td>Hists &amp; Pols</td>\n",
       "      <td>H3K4me1</td>\n",
       "      <td>13:32314880-32315360</td>\n",
       "      <td>ENCODE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A549</td>\n",
       "      <td>Hists &amp; Pols</td>\n",
       "      <td>H3K4me1</td>\n",
       "      <td>13:32316300-32316900</td>\n",
       "      <td>ENCODE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A549</td>\n",
       "      <td>Hists &amp; Pols</td>\n",
       "      <td>H3K4me1</td>\n",
       "      <td>13:32317000-32317240</td>\n",
       "      <td>ENCODE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Cell type  Evidence type Feature name              Location  Source\n",
       "0      A549  DNase1 & TFBS       DNase1  13:32315692-32316237  ENCODE\n",
       "1      A549   Hists & Pols      H3K4me3  13:32314785-32316918  ENCODE\n",
       "2      A549   Hists & Pols      H3K4me1  13:32314880-32315360  ENCODE\n",
       "3      A549   Hists & Pols      H3K4me1  13:32316300-32316900  ENCODE\n",
       "4      A549   Hists & Pols      H3K4me1  13:32317000-32317240  ENCODE"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df4 = pd.read_csv('../data/Evidence-Homo_sapiens_Regulation_Evidence_ENSR00000060894.csv')\n",
    "df4.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['40', '50', '60', '70', '80', '90', '100', '110', '120', '130', '140',\n",
      "       '150', '160', '170', '180', '190', '200', '210', '220', '230', '240',\n",
      "       '250', '260'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>40</th>\n",
       "      <th>50</th>\n",
       "      <th>60</th>\n",
       "      <th>70</th>\n",
       "      <th>80</th>\n",
       "      <th>90</th>\n",
       "      <th>100</th>\n",
       "      <th>110</th>\n",
       "      <th>120</th>\n",
       "      <th>130</th>\n",
       "      <th>...</th>\n",
       "      <th>170</th>\n",
       "      <th>180</th>\n",
       "      <th>190</th>\n",
       "      <th>200</th>\n",
       "      <th>210</th>\n",
       "      <th>220</th>\n",
       "      <th>230</th>\n",
       "      <th>240</th>\n",
       "      <th>250</th>\n",
       "      <th>260</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>time</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>YAL001C</th>\n",
       "      <td>-0.070</td>\n",
       "      <td>-0.23</td>\n",
       "      <td>-0.100</td>\n",
       "      <td>0.03</td>\n",
       "      <td>-0.04</td>\n",
       "      <td>-0.12</td>\n",
       "      <td>-0.28</td>\n",
       "      <td>-0.44</td>\n",
       "      <td>-0.09</td>\n",
       "      <td>0.12</td>\n",
       "      <td>...</td>\n",
       "      <td>0.59</td>\n",
       "      <td>0.34</td>\n",
       "      <td>-0.28</td>\n",
       "      <td>-0.09</td>\n",
       "      <td>-0.44</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YAL014C</th>\n",
       "      <td>0.215</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.025</td>\n",
       "      <td>-0.04</td>\n",
       "      <td>-0.04</td>\n",
       "      <td>-0.02</td>\n",
       "      <td>-0.51</td>\n",
       "      <td>-0.08</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.46</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.30</td>\n",
       "      <td>-0.38</td>\n",
       "      <td>0.07</td>\n",
       "      <td>-0.04</td>\n",
       "      <td>0.13</td>\n",
       "      <td>-0.06</td>\n",
       "      <td>-0.26</td>\n",
       "      <td>-0.10</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YAL016W</th>\n",
       "      <td>0.150</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.220</td>\n",
       "      <td>0.29</td>\n",
       "      <td>-0.10</td>\n",
       "      <td>0.15</td>\n",
       "      <td>-0.73</td>\n",
       "      <td>0.19</td>\n",
       "      <td>-0.15</td>\n",
       "      <td>0.29</td>\n",
       "      <td>...</td>\n",
       "      <td>0.12</td>\n",
       "      <td>-0.17</td>\n",
       "      <td>0.11</td>\n",
       "      <td>-0.15</td>\n",
       "      <td>0.03</td>\n",
       "      <td>-0.26</td>\n",
       "      <td>-0.34</td>\n",
       "      <td>-0.34</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YAL020C</th>\n",
       "      <td>-0.350</td>\n",
       "      <td>-0.28</td>\n",
       "      <td>-0.215</td>\n",
       "      <td>-0.15</td>\n",
       "      <td>0.16</td>\n",
       "      <td>-0.12</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.13</td>\n",
       "      <td>-0.20</td>\n",
       "      <td>...</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.61</td>\n",
       "      <td>-0.20</td>\n",
       "      <td>0.49</td>\n",
       "      <td>-0.43</td>\n",
       "      <td>0.80</td>\n",
       "      <td>-0.47</td>\n",
       "      <td>1.01</td>\n",
       "      <td>-0.36</td>\n",
       "      <td>-0.405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YAL022C</th>\n",
       "      <td>-0.415</td>\n",
       "      <td>-0.59</td>\n",
       "      <td>-0.580</td>\n",
       "      <td>-0.57</td>\n",
       "      <td>-0.09</td>\n",
       "      <td>-0.34</td>\n",
       "      <td>0.49</td>\n",
       "      <td>0.32</td>\n",
       "      <td>1.15</td>\n",
       "      <td>0.20</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.48</td>\n",
       "      <td>-0.40</td>\n",
       "      <td>-0.59</td>\n",
       "      <td>0.54</td>\n",
       "      <td>-0.09</td>\n",
       "      <td>1.03</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.57</td>\n",
       "      <td>-0.26</td>\n",
       "      <td>-0.310</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            40    50     60    70    80    90   100   110   120   130  ...    \\\n",
       "time                                                                   ...     \n",
       "YAL001C -0.070 -0.23 -0.100  0.03 -0.04 -0.12 -0.28 -0.44 -0.09  0.12  ...     \n",
       "YAL014C  0.215  0.09  0.025 -0.04 -0.04 -0.02 -0.51 -0.08  0.00  0.46  ...     \n",
       "YAL016W  0.150  0.15  0.220  0.29 -0.10  0.15 -0.73  0.19 -0.15  0.29  ...     \n",
       "YAL020C -0.350 -0.28 -0.215 -0.15  0.16 -0.12  0.26  0.00  0.13 -0.20  ...     \n",
       "YAL022C -0.415 -0.59 -0.580 -0.57 -0.09 -0.34  0.49  0.32  1.15  0.20  ...     \n",
       "\n",
       "          170   180   190   200   210   220   230   240   250    260  \n",
       "time                                                                  \n",
       "YAL001C  0.59  0.34 -0.28 -0.09 -0.44  0.31  0.03  0.57  0.00  0.010  \n",
       "YAL014C -0.30 -0.38  0.07 -0.04  0.13 -0.06 -0.26 -0.10  0.27  0.235  \n",
       "YAL016W  0.12 -0.17  0.11 -0.15  0.03 -0.26 -0.34 -0.34  0.25  0.190  \n",
       "YAL020C  0.07  0.61 -0.20  0.49 -0.43  0.80 -0.47  1.01 -0.36 -0.405  \n",
       "YAL022C -0.48 -0.40 -0.59  0.54 -0.09  1.03  0.08  0.57 -0.26 -0.310  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df5 = pd.read_csv('../data/Spellman.csv', index_col=0)\n",
    "print(df5.columns)\n",
    "df5.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Spectrum Tag</th>\n",
       "      <th>Spectrum#</th>\n",
       "      <th>Peak#</th>\n",
       "      <th>Intensity</th>\n",
       "      <th>Substance.Mass</th>\n",
       "      <th>Signal/Noise</th>\n",
       "      <th>MZ</th>\n",
       "      <th>MZArea</th>\n",
       "      <th>MZResolution</th>\n",
       "      <th>MZWidthAtHH</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>W924A-QC NAF</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>42.985520</td>\n",
       "      <td>4070.767658</td>\n",
       "      <td>43.102095</td>\n",
       "      <td>4071.774934</td>\n",
       "      <td>656.330526</td>\n",
       "      <td>349.882424</td>\n",
       "      <td>11.637552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>W924A-QC NAF</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>15.286236</td>\n",
       "      <td>4203.107574</td>\n",
       "      <td>15.830504</td>\n",
       "      <td>4204.114850</td>\n",
       "      <td>389.215641</td>\n",
       "      <td>274.219880</td>\n",
       "      <td>15.331182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>W924A-QC NAF</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>5.994020</td>\n",
       "      <td>5159.249115</td>\n",
       "      <td>7.994052</td>\n",
       "      <td>5160.256391</td>\n",
       "      <td>129.746427</td>\n",
       "      <td>214.028077</td>\n",
       "      <td>24.110184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>W924A-QC NAF</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>2.458047</td>\n",
       "      <td>7534.229768</td>\n",
       "      <td>8.585557</td>\n",
       "      <td>7535.237045</td>\n",
       "      <td>131.583387</td>\n",
       "      <td>35.869196</td>\n",
       "      <td>210.075438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>W924A-QC NAF</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>0.793117</td>\n",
       "      <td>13679.509883</td>\n",
       "      <td>14.219609</td>\n",
       "      <td>13680.517159</td>\n",
       "      <td>255.481347</td>\n",
       "      <td>87.503297</td>\n",
       "      <td>156.342876</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Spectrum Tag  Spectrum#  Peak#  Intensity  Substance.Mass  Signal/Noise  \\\n",
       "0  W924A-QC NAF          1      1  42.985520     4070.767658     43.102095   \n",
       "1  W924A-QC NAF          1      2  15.286236     4203.107574     15.830504   \n",
       "2  W924A-QC NAF          1      3   5.994020     5159.249115      7.994052   \n",
       "3  W924A-QC NAF          1      4   2.458047     7534.229768      8.585557   \n",
       "4  W924A-QC NAF          1      5   0.793117    13679.509883     14.219609   \n",
       "\n",
       "             MZ      MZArea  MZResolution  MZWidthAtHH  \n",
       "0   4071.774934  656.330526    349.882424    11.637552  \n",
       "1   4204.114850  389.215641    274.219880    15.331182  \n",
       "2   5160.256391  129.746427    214.028077    24.110184  \n",
       "3   7535.237045  131.583387     35.869196   210.075438  \n",
       "4  13680.517159  255.481347     87.503297   156.342876  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df6 = pd.read_csv('../data/kuererqcPeaksDefault.csv')\n",
    "df6.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Transcript ID</th>\n",
       "      <th>bp</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Translation ID</th>\n",
       "      <th>Biotype</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>UniProt</th>\n",
       "      <th>RefSeq</th>\n",
       "      <th>Flags</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRCA2-001</td>\n",
       "      <td>ENST00000380152.7</td>\n",
       "      <td>11986</td>\n",
       "      <td>3418aa</td>\n",
       "      <td>ENSP00000369497</td>\n",
       "      <td>Protein coding</td>\n",
       "      <td>CCDS9344</td>\n",
       "      <td>P51587</td>\n",
       "      <td>-</td>\n",
       "      <td>TSL:5GENCODE basicAPPRIS P1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2-201</td>\n",
       "      <td>ENST00000544455.5</td>\n",
       "      <td>10984</td>\n",
       "      <td>3418aa</td>\n",
       "      <td>ENSP00000439902</td>\n",
       "      <td>Protein coding</td>\n",
       "      <td>CCDS9344</td>\n",
       "      <td>P51587</td>\n",
       "      <td>NM_000059  NP_000050</td>\n",
       "      <td>TSL:1GENCODE basicAPPRIS P1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRCA2-002</td>\n",
       "      <td>ENST00000470094.1</td>\n",
       "      <td>842</td>\n",
       "      <td>186aa</td>\n",
       "      <td>ENSP00000434898</td>\n",
       "      <td>Nonsense mediated decay</td>\n",
       "      <td>-</td>\n",
       "      <td>H0YE37</td>\n",
       "      <td>-</td>\n",
       "      <td>CDS 5' incompleteTSL:5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRCA2-005</td>\n",
       "      <td>ENST00000528762.1</td>\n",
       "      <td>495</td>\n",
       "      <td>64aa</td>\n",
       "      <td>ENSP00000433168</td>\n",
       "      <td>Nonsense mediated decay</td>\n",
       "      <td>-</td>\n",
       "      <td>H0YD86</td>\n",
       "      <td>-</td>\n",
       "      <td>CDS 5' incompleteTSL:4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRCA2-004</td>\n",
       "      <td>ENST00000614259.1</td>\n",
       "      <td>7950</td>\n",
       "      <td>No protein</td>\n",
       "      <td>-</td>\n",
       "      <td>Processed transcript</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>TSL:2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Name      Transcript ID     bp     Protein   Translation ID  \\\n",
       "0  BRCA2-001  ENST00000380152.7  11986      3418aa  ENSP00000369497   \n",
       "1  BRCA2-201  ENST00000544455.5  10984      3418aa  ENSP00000439902   \n",
       "2  BRCA2-002  ENST00000470094.1    842       186aa  ENSP00000434898   \n",
       "3  BRCA2-005  ENST00000528762.1    495        64aa  ENSP00000433168   \n",
       "4  BRCA2-004  ENST00000614259.1   7950  No protein                -   \n",
       "\n",
       "                    Biotype      CCDS UniProt                RefSeq  \\\n",
       "0            Protein coding  CCDS9344  P51587                     -   \n",
       "1            Protein coding  CCDS9344  P51587  NM_000059  NP_000050   \n",
       "2   Nonsense mediated decay         -  H0YE37                     -   \n",
       "3   Nonsense mediated decay         -  H0YD86                     -   \n",
       "4      Processed transcript         -       -                     -   \n",
       "\n",
       "                         Flags  \n",
       "0  TSL:5GENCODE basicAPPRIS P1  \n",
       "1  TSL:1GENCODE basicAPPRIS P1  \n",
       "2       CDS 5' incompleteTSL:5  \n",
       "3       CDS 5' incompleteTSL:4  \n",
       "4                        TSL:2  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df7 = pd.read_csv('../data/transcripts-Summary-Homo_sapiens_Gene_Summary_ENSG00000139618.csv')\n",
    "df7.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OTU0</th>\n",
       "      <th>OTU1</th>\n",
       "      <th>OTU2</th>\n",
       "      <th>OTU3</th>\n",
       "      <th>OTU4</th>\n",
       "      <th>OTU5</th>\n",
       "      <th>OTU6</th>\n",
       "      <th>OTU7</th>\n",
       "      <th>OTU8</th>\n",
       "      <th>OTU9</th>\n",
       "      <th>...</th>\n",
       "      <th>OTU6686</th>\n",
       "      <th>OTU6687</th>\n",
       "      <th>OTU6688</th>\n",
       "      <th>OTU6689</th>\n",
       "      <th>OTU6690</th>\n",
       "      <th>OTU6691</th>\n",
       "      <th>OTU6692</th>\n",
       "      <th>OTU6693</th>\n",
       "      <th>OTU6694</th>\n",
       "      <th>OTU6695</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.558520e-11</td>\n",
       "      <td>4.724120e-11</td>\n",
       "      <td>1.225880e-11</td>\n",
       "      <td>4.515660e-11</td>\n",
       "      <td>2.718290e-11</td>\n",
       "      <td>2.706710e-11</td>\n",
       "      <td>3.306490e-11</td>\n",
       "      <td>5.078730e-11</td>\n",
       "      <td>1.845710e-11</td>\n",
       "      <td>0.002566</td>\n",
       "      <td>...</td>\n",
       "      <td>6.661360e-11</td>\n",
       "      <td>3.021640e-11</td>\n",
       "      <td>4.415210e-11</td>\n",
       "      <td>7.311330e-11</td>\n",
       "      <td>7.772160e-11</td>\n",
       "      <td>4.333640e-11</td>\n",
       "      <td>5.437370e-11</td>\n",
       "      <td>8.723940e-11</td>\n",
       "      <td>2.713500e-11</td>\n",
       "      <td>4.966000e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2.359220e-11</td>\n",
       "      <td>9.533200e-11</td>\n",
       "      <td>3.330680e-11</td>\n",
       "      <td>2.669130e-11</td>\n",
       "      <td>2.019270e-11</td>\n",
       "      <td>2.505580e-11</td>\n",
       "      <td>6.753770e-11</td>\n",
       "      <td>9.076440e-11</td>\n",
       "      <td>8.707380e-11</td>\n",
       "      <td>0.000252</td>\n",
       "      <td>...</td>\n",
       "      <td>3.255580e-12</td>\n",
       "      <td>5.385290e-11</td>\n",
       "      <td>4.725960e-11</td>\n",
       "      <td>2.600370e-11</td>\n",
       "      <td>4.242660e-11</td>\n",
       "      <td>6.546480e-11</td>\n",
       "      <td>4.848670e-11</td>\n",
       "      <td>8.377660e-11</td>\n",
       "      <td>3.498560e-11</td>\n",
       "      <td>7.617720e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6.771010e-11</td>\n",
       "      <td>3.682960e-11</td>\n",
       "      <td>8.018540e-11</td>\n",
       "      <td>5.488950e-11</td>\n",
       "      <td>1.341180e-11</td>\n",
       "      <td>5.115210e-11</td>\n",
       "      <td>5.806820e-11</td>\n",
       "      <td>6.466670e-11</td>\n",
       "      <td>1.447080e-11</td>\n",
       "      <td>0.002336</td>\n",
       "      <td>...</td>\n",
       "      <td>7.231020e-11</td>\n",
       "      <td>6.298250e-12</td>\n",
       "      <td>7.062920e-11</td>\n",
       "      <td>8.313000e-11</td>\n",
       "      <td>3.307660e-11</td>\n",
       "      <td>7.383310e-11</td>\n",
       "      <td>4.448790e-11</td>\n",
       "      <td>9.645520e-12</td>\n",
       "      <td>7.880280e-11</td>\n",
       "      <td>3.992640e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5.520270e-11</td>\n",
       "      <td>9.890760e-11</td>\n",
       "      <td>4.581380e-11</td>\n",
       "      <td>3.539770e-11</td>\n",
       "      <td>2.092110e-11</td>\n",
       "      <td>6.255510e-11</td>\n",
       "      <td>9.341670e-11</td>\n",
       "      <td>5.372400e-11</td>\n",
       "      <td>6.280840e-11</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>...</td>\n",
       "      <td>9.095820e-11</td>\n",
       "      <td>2.935560e-11</td>\n",
       "      <td>9.996590e-11</td>\n",
       "      <td>9.821790e-11</td>\n",
       "      <td>8.541320e-11</td>\n",
       "      <td>9.727770e-11</td>\n",
       "      <td>2.959570e-11</td>\n",
       "      <td>8.581660e-13</td>\n",
       "      <td>5.876120e-11</td>\n",
       "      <td>5.506230e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5.238120e-11</td>\n",
       "      <td>6.340330e-11</td>\n",
       "      <td>2.354280e-11</td>\n",
       "      <td>7.469650e-11</td>\n",
       "      <td>2.489830e-11</td>\n",
       "      <td>4.048870e-13</td>\n",
       "      <td>3.337430e-11</td>\n",
       "      <td>8.499950e-12</td>\n",
       "      <td>4.558400e-11</td>\n",
       "      <td>0.000493</td>\n",
       "      <td>...</td>\n",
       "      <td>1.504300e-11</td>\n",
       "      <td>4.900990e-11</td>\n",
       "      <td>8.106680e-12</td>\n",
       "      <td>4.669770e-11</td>\n",
       "      <td>8.265580e-11</td>\n",
       "      <td>4.631170e-11</td>\n",
       "      <td>1.600100e-11</td>\n",
       "      <td>5.546980e-11</td>\n",
       "      <td>7.840300e-11</td>\n",
       "      <td>8.564610e-11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 6696 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           OTU0          OTU1          OTU2          OTU3          OTU4  \\\n",
       "0  1.558520e-11  4.724120e-11  1.225880e-11  4.515660e-11  2.718290e-11   \n",
       "1  2.359220e-11  9.533200e-11  3.330680e-11  2.669130e-11  2.019270e-11   \n",
       "2  6.771010e-11  3.682960e-11  8.018540e-11  5.488950e-11  1.341180e-11   \n",
       "3  5.520270e-11  9.890760e-11  4.581380e-11  3.539770e-11  2.092110e-11   \n",
       "4  5.238120e-11  6.340330e-11  2.354280e-11  7.469650e-11  2.489830e-11   \n",
       "\n",
       "           OTU5          OTU6          OTU7          OTU8      OTU9  \\\n",
       "0  2.706710e-11  3.306490e-11  5.078730e-11  1.845710e-11  0.002566   \n",
       "1  2.505580e-11  6.753770e-11  9.076440e-11  8.707380e-11  0.000252   \n",
       "2  5.115210e-11  5.806820e-11  6.466670e-11  1.447080e-11  0.002336   \n",
       "3  6.255510e-11  9.341670e-11  5.372400e-11  6.280840e-11  0.000741   \n",
       "4  4.048870e-13  3.337430e-11  8.499950e-12  4.558400e-11  0.000493   \n",
       "\n",
       "       ...            OTU6686       OTU6687       OTU6688       OTU6689  \\\n",
       "0      ...       6.661360e-11  3.021640e-11  4.415210e-11  7.311330e-11   \n",
       "1      ...       3.255580e-12  5.385290e-11  4.725960e-11  2.600370e-11   \n",
       "2      ...       7.231020e-11  6.298250e-12  7.062920e-11  8.313000e-11   \n",
       "3      ...       9.095820e-11  2.935560e-11  9.996590e-11  9.821790e-11   \n",
       "4      ...       1.504300e-11  4.900990e-11  8.106680e-12  4.669770e-11   \n",
       "\n",
       "        OTU6690       OTU6691       OTU6692       OTU6693       OTU6694  \\\n",
       "0  7.772160e-11  4.333640e-11  5.437370e-11  8.723940e-11  2.713500e-11   \n",
       "1  4.242660e-11  6.546480e-11  4.848670e-11  8.377660e-11  3.498560e-11   \n",
       "2  3.307660e-11  7.383310e-11  4.448790e-11  9.645520e-12  7.880280e-11   \n",
       "3  8.541320e-11  9.727770e-11  2.959570e-11  8.581660e-13  5.876120e-11   \n",
       "4  8.265580e-11  4.631170e-11  1.600100e-11  5.546980e-11  7.840300e-11   \n",
       "\n",
       "        OTU6695  \n",
       "0  4.966000e-11  \n",
       "1  7.617720e-11  \n",
       "2  3.992640e-11  \n",
       "3  5.506230e-11  \n",
       "4  8.564610e-11  \n",
       "\n",
       "[5 rows x 6696 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df8 = pd.read_csv('../data/Microbiome.csv')\n",
    "df8.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'CollectionMet': {'0': 'Contents', '1': 'Scraping'},\n",
       " 'Diet': {'0': 'LFPP',\n",
       "  '1': 'Western',\n",
       "  '2': 'CARBR',\n",
       "  '3': 'FATR',\n",
       "  '4': 'Suckling',\n",
       "  '5': 'Human'},\n",
       " 'Donor': {'0': 'HMouseLFPP',\n",
       "  '1': 'CONVR',\n",
       "  '2': 'Human',\n",
       "  '3': 'Fresh',\n",
       "  '4': 'Frozen',\n",
       "  '5': 'HMouseWestern',\n",
       "  '6': 'CONVD'},\n",
       " 'Sex': {'0': 'Male', '1': 'Female'},\n",
       " 'Source': {'0': 'Cecum1',\n",
       "  '1': 'Cecum2',\n",
       "  '10': 'SI9',\n",
       "  '11': 'Stomach',\n",
       "  '12': 'Cecum',\n",
       "  '2': 'Colon1',\n",
       "  '3': 'Colon2',\n",
       "  '4': 'Feces',\n",
       "  '5': 'SI1',\n",
       "  '6': 'SI13',\n",
       "  '7': 'SI15',\n",
       "  '8': 'SI2',\n",
       "  '9': 'SI5'}}"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import re\n",
    "df9 = {}\n",
    "with open('../data/MicrobiomeMetadataDictionary.csv') as file:\n",
    "    righe = file.readlines()\n",
    "\n",
    "for riga in righe:\n",
    "    m = re.search(\"(.+),$\", riga)\n",
    "    if m:\n",
    "        tabella = {}\n",
    "        df9[m.group(1)] = tabella\n",
    "    m = re.search(\"(.+),(.+)$\", riga)\n",
    "    if m:\n",
    "        tabella[m.group(1)] = m.group(2)\n",
    "    m = re.search(\"^,$\", riga)\n",
    "\n",
    "df9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2+"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
